scripts/check-recovery.js

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable no-console */
+import parser from "solidity-parser-antlr";
+import fs from "fs";
+import path from "path";
+
+// Taken from https://gist.github.com/kethinov/6658166#gistcomment-1941504
+const walkSync = (dir, filelist = []) => {
+  fs.readdirSync(dir).forEach(file => {
+    filelist = fs.statSync(path.join(dir, file)).isDirectory() ? walkSync(path.join(dir, file), filelist) : filelist.concat(path.join(dir, file)); // eslint-disable-line no-param-reassign
+  });
+  return filelist;
+};
+
+function correctRecoveryModifier(functionDef) {
+  const isPrivate = ["private", "internal"].indexOf(functionDef.visibility) > -1;
+  const isView = ["view", "pure"].indexOf(functionDef.stateMutability) > -1;
+  const hasModifier = functionDef.modifiers.filter(mod => ["stoppable", "recovery"].indexOf(mod.name) > -1).length > 0;
+  return isPrivate || (isView || hasModifier);
+}
+
+walkSync("./contracts/").forEach(contractName => {
+  // Only these contracts are involved in recovery mode
+  if (["contracts/Colony.sol", "contracts/ColonyFunding.sol", "contracts/ColonyTask.sol"].indexOf(contractName) < 0) {
```

Comment:
I think this script would work better in the future if we list the paths that it shouldn't run over, like I currently do in the storage variables script. If we add a new contract to Colony (which we surely will in the future), we have to remember to add it here, rather than being reminded to if it's implemented the other way and we forget.

Reply:
That's reasonable. I went this way because relatively few contracts need to be checked, but I see how the "opt-out" approach will make it harder to screw up down the road.